lib/services/search_service.dart

```diff
@@ -0,0 +1,180 @@
+    Future.delayed(const Duration(seconds: 5), () async {
+      // In case home screen loads before 5 seconds and user starts search, future will not be null
+      _future == null
+          ? getAllFiles().then((value) {
+              _cachedFiles = value;
+            })
+          : null;
+    });
+
+    Bus.instance.on<LocalPhotosUpdatedEvent>().listen((event) {
+      _cachedFiles = null;
+      getAllFiles();
+    });
+  }
+
+  Future<List<File>> getAllFiles() async {
+    if (_cachedFiles != null) {
+      return _cachedFiles;
+    }
+    if (_future != null) {
+      return _future;
+    }
+    _future = _fetchAllFiles();
+    return _future;
+  }
+
+  Future<List<File>> _fetchAllFiles() async {
+    _cachedFiles = await FilesDB.instance.getAllFilesFromDB();
+    return _cachedFiles;
+  }
+
+  Future<List<File>> getFilesOnFilenameSearch(String query) async {
+    final List<File> matchedFiles = [];
+    final List<File> files = await getAllFiles();
+    for (int i = 0;
+        (i < files.length) && (matchedFiles.length < _maximumResultsLimit);
+        i++) {
+      File file = files[i];
+      if (file.title.contains(RegExp(query, caseSensitive: false))) {
+        matchedFiles.add(file);
+      }
+    }
+    return matchedFiles;
+  }
+
+  void clearCache() {
+    _cachedFiles.clear();
+  }
+
+  Future<List<LocationSearchResult>> getLocationsAndMatchedFiles(
+    String query,
+  ) async {
+    try {
+      final List<File> allFiles = await SearchService.instance.getAllFiles();
+      final List<LocationSearchResult> locationsAndMatchedFiles = [];
+
+      final response = await _dio.get(
+        _config.getHttpEndpoint() + "/search/location",
+        queryParameters: {"query": query, "limit": 4},
+        options: Options(
+          headers: {"X-Auth-Token": _config.getToken()},
+        ),
+      );
+
+      final matchedLocationNamesAndBboxs =
+          ResultsToListOfPlaceAndBbox.fromMap(response.data);
+
+      for (PlaceAndBbox locationAndBbox
+          in matchedLocationNamesAndBboxs.results) {
+        locationsAndMatchedFiles.add(
+          LocationSearchResult(locationAndBbox.place, []),
+        );
+        for (File file in allFiles) {
+          if (_isValidLocation(file.location)) {
+            //format returned by the api is [lng,lat,lng,lat] where indexes 0 & 1 are southwest and 2 & 3 northeast
+            if (file.location.longitude > locationAndBbox.bbox[0] &&
+                file.location.latitude > locationAndBbox.bbox[1] &&
+                file.location.longitude < locationAndBbox.bbox[2] &&
+                file.location.latitude < locationAndBbox.bbox[3]) {
+              locationsAndMatchedFiles.last.files.add(file);
+            }
+          }
+        }
+      }
+      locationsAndMatchedFiles.removeWhere((e) => e.files.isEmpty);
+      return locationsAndMatchedFiles;
+    } on DioError catch (e) {
+      _logger.info(e);
+      rethrow;
+    }
+  }
+
+  bool _isValidLocation(Location location) {
```

Comment:
It's always better to keep public functions at the top, since it will let us glance at the sidebar to get an idea about the public API a class exposes. Perhaps there's a way to get the linter to do this.